### PR TITLE
refactor: Loadable no longer declares onGameResize

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Mixin `Loadable` no longer declares method `onGameResize()`. Instead, this method is now defined by classes `Component` and `Game` separately (which they were already doing before).

Resizing has nothing to do with loading, so it is not proper to declare this method inside `Loadable`. I presume it existed there by accident. 

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- (NA) I have updated/added tests for ALL new/updated/fixed functionality.
- (NA) I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- (NA) I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/lib/src/game/flame_game.dart [hunk 1]
```diff
@@ -106,7 +106,11 @@ class FlameGame extends Component with Game {
   @mustCallSuper
   void onGameResize(Vector2 canvasSize) {
     camera.handleResize(canvasSize);
-    super.onGameResize(canvasSize);
+    super.onGameResize(canvasSize); // Game.onGameResize
+    // [onGameResize] is declared both in [Component] and in [Game]. Since
+    // there is no way to explicitly call the [Component]'s implementation,
+    // we propagate the event to [FlameGame]'s children manually.
+    children.forEach((child) => child.onGameResize(canvasSize));
```
Comment (spydon) [inline packages/flame/lib/src/game/flame_game.dart:113]: It feels like this makes it worse since this code duplication wasn't needed before.
It is resized when loaded, therefore one could argue that it can live in `Loadable`. A bit confusing maybe, but I think still better than code duplication.

Comment (st-pasha) [reply packages/flame/lib/src/game/flame_game.dart:113]: One line of code is barely "code duplication", and it's only needed because of the unfortunate fact that FlameGame attempts to inherit both from Game and from Component. If you're worried about lines of code, you'll notice that this PR removes 6 lines of code while adding only 1.

> It is resized when loaded, therefore one could argue that it can live in Loadable.

It is resized after being loaded too, so one could argue it must not live in Loadable. Regardless, resizing is an implementation detail. We've discussed elsewhere about changing or removing these events completely. While we have not decided whether we want to actually do that, the fact that it is perfectly possible demonstrates that resizing has nothing to do with loading.

**Comment by st-pasha** [comment]
The important benefit that this brings is it makes the code understandable.
Before, we had `Component.onGameResize()` calling `super.onGameResize`, which was empty -- but removing that super call would break the engine. This kind of shady code execution pattern that an IDE cannot follow, should be avoided.

File: packages/flame/lib/src/game/flame_game.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/lib/src/game/flame_game.dart:113]: I guess it is fine since you explain it very well in the comment above, and that line should very seldom change.

**Review by spydon — APPROVED** [review]
This is one of the key points where OOP fails I think, that you want multiple inheritance, but you can't have pure multiple inheritance so it becomes this non-linear pattern which can be hard to follow. I think you are right, this does improve the code, approved!
